bridge/src/transactions/pre_signed_musig2.rs

```diff
@@ -124,7 +124,7 @@ pub fn pre_sign_musig2_taproot_input<T: PreSignedTransaction + PreSignedMusig2Tr
     let script = &tx.prev_scripts()[input_index].clone();
     let musig2_nonces = &tx.musig2_nonces()[&input_index].values().cloned().collect();
 
-    let partial_signature = generate_taproot_partial_signature(
+    let partial_signature = match generate_taproot_partial_signature(
```

Comment:
if we return before updating the witness, but there is no way to catch this exception. 

Besides, there is an `expect` in `generate_taproot_partial_signature`,  my suggestion would be panic immediately and throw the reason.